pkg/liqoctl/status/pods.go

```diff
@@ -0,0 +1,397 @@
+			pc.errors = true
+		}
+	}
+
+	for _, dName := range pc.daemonSets {
+		err := pc.daemonSetStatus(ctx, dName)
+		if err != nil {
+			pc.addCollectionError("DaemonSet", dName, fmt.Errorf("unable to collect status for daemonSet %s in namespace %s: %w", dName, pc.namespace, err))
+			pc.errors = true
+		}
+	}
+
+	return nil
+}
+
+// Format implements the format method of the Checker interface.
+// it outputs the status of the Liqo components in a string ready to be
+// printed out.
+func (pc *podChecker) Format() (string, error) {
+	var buf bytes.Buffer
+	w := tabwriter.NewWriter(&buf, 0, 0, 4, ' ', 0)
+
+	separator := strings.Repeat("-", len(pc.name))
+	fmt.Fprintf(w, "%s\n", pc.name)
+	fmt.Fprintf(w, "%s\n", separator)
+	if pc.errors {
+		// Add pod state to the buffer for each deployment type.
+		if pc.errors {
+			fmt.Fprintf(w, "%s liqo control plane is not OK\n", redCross)
+			for deployment, podState := range pc.podsState {
+				if podState.errorFlag {
+					fmt.Fprintf(w, "%s\t%s\t%s\n", podState.deploymentType, deployment, podState.format())
+					for pod, errorCol := range podState.errors {
+						for _, err := range errorCol.errors {
+							fmt.Fprintf(w, "%s\t%s\tPod\t%s\t%s\n", podState.deploymentType, deployment, pod, err)
+						}
+					}
+				}
+			}
+			for _, err := range pc.collectionErrors {
+				fmt.Fprintf(w, "%s\t%s\t%s\n", err.appType, err.appName, err.err)
+			}
+		}
+	} else {
+		fmt.Fprintf(w, "%s control plane pods are up and running\n", checkMark)
+	}
+
+	if err := w.Flush(); err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+// deploymentStatus collects the status of a given kubernetes Deployment.
+func (pc *podChecker) deploymentStatus(ctx context.Context, deploymentName string) error {
+	var errors bool
+	d, err := pc.client.AppsV1().Deployments(pc.namespace).Get(ctx, deploymentName, metav1.GetOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	if d == nil {
+		return fmt.Errorf("deployment %s seems to be unavailable", deploymentName)
+	}
+
+	dState := newPodState("Deployment")
+	dState.desired = int(d.Status.Replicas)
+	dState.ready = int(d.Status.ReadyReplicas)
+	dState.unavailable = int(d.Status.UnavailableReplicas)
+	dState.available = int(d.Status.AvailableReplicas)
+
+	// Get all the pods related with the current deployment.
+	pods, err := pc.client.CoreV1().Pods(pc.namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labels.FormatLabels(d.Spec.Selector.MatchLabels),
+	})
+	if err != nil {
+		return err
+	}
+	if len(pods.Items) == 0 {
+		return fmt.Errorf("no pods found for deployment %s", deploymentName)
+	}
+
+	if errors = checkPodsStatus(pods.Items, &dState); errors {
+		pc.errors = errors
+	}
+	pc.podsState[deploymentName] = dState
+
+	return nil
+}
+
+// daemontSetStatus collects the status of a given kubernetes DaemonSet.
+func (pc *podChecker) daemonSetStatus(ctx context.Context, daemonSetName string) error {
```

Comment:
I would add a warning in case the number of DaemonSet pods is less than the number of non-virtual Nodes

This could happen in the case of taints over some nodes, and applications running on these nodes will not be able to join the liqo networking

Reply:
I was planning to add it in the next PR.

Reply:
👍